aesd-char-driver/aesd-circular-buffer.c

```diff
@@ -89,3 +91,23 @@ void aesd_circular_buffer_init(struct aesd_circular_buffer *buffer)
 {
     memset(buffer,0,sizeof(struct aesd_circular_buffer));
 }
+
+
+void aesd_circular_buffer_cleanup(struct aesd_circular_buffer *buffer)
+{
+	uint8_t index;
+	struct aesd_buffer_entry *entry;
+
+	AESD_CIRCULAR_BUFFER_FOREACH(entry,buffer,index) 
+	{
+		if (entry->buffptr == NULL) continue;
+		
+
+#ifdef __KERNEL__
+		kfree(entry->buffptr);
```

Comment:
@cchoi22915 I'd put the logic to free buffer in the application rather than this function, since the application will know whether the buffer was dynamically allocated.